File: Question2/stack.js
class Stack {
	constructor() {
		this.stack = [];
	}

	find(element) {
		return this.stack.indexOf(element);
	}

	push(element) {
		this.stack.push(element);
	}

	pop() {
		return this.stack.pop();
	}

	size() {
		return this.stack.length;
	}

	isEmpty() {
		return this.stack.length === 0;
	}
}

File: Question2/stack.lua
Stack = {}
Stack.__index = Stack

function Stack.new()
  local stack = {}
  setmetatable(stack, Stack)
  stack.items = {}
  return stack
end

function Stack:find(element)
  for i, item in ipairs(self.items) do
    if item == element then
      return i
    end
  end
  return nil
end

function Stack:push(element)
  table.insert(self.items, element)
end

function Stack:pop()
  return table.remove(self.items)
end

function Stack:size()
  return #self.items
end

function Stack:isEmpty()
  return self:size() == 0
end